点击 ’手动执行签到‘ 无效

点击 ’手动执行签到‘ 跳到，日志页面，然后就空白，没然后，也没有执行定时签到

Wed Feb  3 18:09:46 2021 daemon.err uhttpd[3654]: sh: /usr/share/jd-dailybonus/newapp.sh: Permission denied

> node /usr/share/jd-dailybonus/JD_DailyBonus.js，，用这个，应该能看到点错误代码，应该还是node版本的过，我从0.85一直升到1.02都是这个问题，然后把node升级到最新版本，中间还安装过几个低版本，还是不能用，装回0.85也不能用了，装回当初的node还是不能用，最后把路由彻底恢复最初状态，0.85能用了。也可能和我用的系统内核底有关吧，一直用着这个版本感觉挺稳定，一直就没再折腾，20年6月的版本

不是去掉requset依赖了么？

root@OpenWrt:/# node /usr/share/jd-dailybonus/JD_DailyBonus.js 
internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module 'request'
Require stack:
- /usr/share/jd-dailybonus/JD_DailyBonus.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at /usr/share/jd-dailybonus/JD_DailyBonus.js:2265:23
    at nobyda (/usr/share/jd-dailybonus/JD_DailyBonus.js:2274:5)
    at Object.<anonymous> (/usr/share/jd-dailybonus/JD_DailyBonus.js:103:15)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/usr/share/jd-dailybonus/JD_DailyBonus.js' ]
}

> > node /usr/share/jd-dailybonus/JD_DailyBonus.js，，用这个，应该能看到点错误代码，应该还是node版本的过，我从0.85一直升到1.02都是这个问题，然后把node升级到最新版本，中间还安装过几个低版本，还是不能用，装回0.85也不能用了，装回当初的node还是不能用，最后把路由彻底恢复最初状态，0.85能用了。也可能和我用的系统内核底有关吧，一直用着这个版本感觉挺稳定，一直就没再折腾，20年6月的版本
> 
> 不是去掉requset依赖了么？
> 
> root@OpenWrt:/# node /usr/share/jd-dailybonus/JD_DailyBonus.js
> internal/modules/cjs/loader.js:968
> throw err;
> ^
> 
> Error: Cannot find module 'request'
> Require stack:
> 
> * /usr/share/jd-dailybonus/JD_DailyBonus.js
>   at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
>   at Function.Module._load (internal/modules/cjs/loader.js:841:27)
>   at Module.require (internal/modules/cjs/loader.js:1025:19)
>   at require (internal/modules/cjs/helpers.js:72:18)
>   at /usr/share/jd-dailybonus/JD_DailyBonus.js:2265:23
>   at nobyda (/usr/share/jd-dailybonus/JD_DailyBonus.js:2274:5)
>   at Object. (/usr/share/jd-dailybonus/JD_DailyBonus.js:103:15)
>   at Module._compile (internal/modules/cjs/loader.js:1137:30)
>   at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
>   at Module.load (internal/modules/cjs/loader.js:985:32) {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: [ '/usr/share/jd-dailybonus/JD_DailyBonus.js' ]
>   }

去掉的是lua-request，node的request依赖是上游NobyDa的脚本的依赖，去不掉

> > > node /usr/share/jd-dailybonus/JD_DailyBonus.js，，用这个，应该能看到点错误代码，应该还是node版本的过，我从0.85一直升到1.02都是这个问题，然后把node升级到最新版本，中间还安装过几个低版本，还是不能用，装回0.85也不能用了，装回当初的node还是不能用，最后把路由彻底恢复最初状态，0.85能用了。也可能和我用的系统内核底有关吧，一直用着这个版本感觉挺稳定，一直就没再折腾，20年6月的版本
> > 
> > 
> > 不是去掉requset依赖了么？
> > root@OpenWrt:/# node /usr/share/jd-dailybonus/JD_DailyBonus.js
> > internal/modules/cjs/loader.js:968
> > throw err;
> > ^
> > Error: Cannot find module 'request'
> > Require stack:
> > 
> > * /usr/share/jd-dailybonus/JD_DailyBonus.js
> >   at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
> >   at Function.Module._load (internal/modules/cjs/loader.js:841:27)
> >   at Module.require (internal/modules/cjs/loader.js:1025:19)
> >   at require (internal/modules/cjs/helpers.js:72:18)
> >   at /usr/share/jd-dailybonus/JD_DailyBonus.js:2265:23
> >   at nobyda (/usr/share/jd-dailybonus/JD_DailyBonus.js:2274:5)
> >   at Object. (/usr/share/jd-dailybonus/JD_DailyBonus.js:103:15)
> >   at Module._compile (internal/modules/cjs/loader.js:1137:30)
> >   at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
> >   at Module.load (internal/modules/cjs/loader.js:985:32) {
> >   code: 'MODULE_NOT_FOUND',
> >   requireStack: [ '/usr/share/jd-dailybonus/JD_DailyBonus.js' ]
> >   }
> 
> 去掉的是lua-request，node的request依赖是上游NobyDa的脚本的依赖，去不掉

你的最新修改增加权限后可以使用，国内更新源有问题很慢

> > > > node /usr/share/jd-dailybonus/JD_DailyBonus.js，，用这个，应该能看到点错误代码，应该还是node版本的过，我从0.85一直升到1.02都是这个问题，然后把node升级到最新版本，中间还安装过几个低版本，还是不能用，装回0.85也不能用了，装回当初的node还是不能用，最后把路由彻底恢复最初状态，0.85能用了。也可能和我用的系统内核底有关吧，一直用着这个版本感觉挺稳定，一直就没再折腾，20年6月的版本
> > > 
> > > 
> > > 不是去掉requset依赖了么？
> > > root@OpenWrt:/# node /usr/share/jd-dailybonus/JD_DailyBonus.js
> > > internal/modules/cjs/loader.js:968
> > > throw err;
> > > ^
> > > Error: Cannot find module 'request'
> > > Require stack:
> > > 
> > > * /usr/share/jd-dailybonus/JD_DailyBonus.js
> > >   at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
> > >   at Function.Module._load (internal/modules/cjs/loader.js:841:27)
> > >   at Module.require (internal/modules/cjs/loader.js:1025:19)
> > >   at require (internal/modules/cjs/helpers.js:72:18)
> > >   at /usr/share/jd-dailybonus/JD_DailyBonus.js:2265:23
> > >   at nobyda (/usr/share/jd-dailybonus/JD_DailyBonus.js:2274:5)
> > >   at Object. (/usr/share/jd-dailybonus/JD_DailyBonus.js:103:15)
> > >   at Module._compile (internal/modules/cjs/loader.js:1137:30)
> > >   at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
> > >   at Module.load (internal/modules/cjs/loader.js:985:32) {
> > >   code: 'MODULE_NOT_FOUND',
> > >   requireStack: [ '/usr/share/jd-dailybonus/JD_DailyBonus.js' ]
> > >   }
> > 
> > 
> > 去掉的是lua-request，node的request依赖是上游NobyDa的脚本的依赖，去不掉
> 
> 你的最新修改增加权限后可以使用，国内更新源有问题很慢

在lean lede编译后newapp.sh是有运行权限的,面对三无issue，我只能上根治方案
我一直用的GitHub源，国内更新源大概是gitee的问题，可以试试CDN源

--- a/luasrc/controller/jd-dailybonus.lua
+++ b/luasrc/controller/jd-dailybonus.lua
@@ -22,7 +22,7 @@ end
 function run()
     local running = luci.sys.call("busybox ps -w | grep JD_DailyBonus.js | grep -v grep >/dev/null") == 0
     if not running then
-        luci.sys.call('/usr/share/jd-dailybonus/newapp.sh -r')
+        luci.sys.call('sh /usr/share/jd-dailybonus/newapp.sh -r')
     end
     luci.http.write('')
 end
@@ -31,7 +31,7 @@ end
 function check_update()
     local jd = 'jd-dailybonus'
     local e = {}
-    local new_version = luci.sys.exec('/usr/share/jd-dailybonus/newapp.sh -n')
+    local new_version = luci.sys.exec('sh /usr/share/jd-dailybonus/newapp.sh -n')
     e.new_version = new_version
     e.error = 0
     luci.http.prepare_content('application/json')
@@ -45,7 +45,7 @@ function update()
     local uci = luci.model.uci.cursor()
     local version = luci.http.formvalue('version')
     --下载脚本
-    local code = luci.sys.exec('/usr/share/jd-dailybonus/newapp.sh -u')
+    local code = luci.sys.exec('sh /usr/share/jd-dailybonus/newapp.sh -u')
     e.error = code
     luci.http.prepare_content('application/json')
     luci.http.write_json(e)

--- a/root/etc/init.d/jd-dailybonus
+++ b/root/etc/init.d/jd-dailybonus
@@ -19,7 +19,7 @@ del_cron() {
 }
 
 start_service(){
-	/usr/share/jd-dailybonus/newapp.sh -s
+	sh /usr/share/jd-dailybonus/newapp.sh -s
 }
 
 stop_service() {

--- a/root/usr/share/jd-dailybonus/newapp.sh
+++ b/root/usr/share/jd-dailybonus/newapp.sh
@@ -51,8 +51,8 @@ cancel() {
 
 add_cron() {
     sed -i '/jd-dailybonus/d' $CRON_FILE
-    [ $(uci_get_by_type global auto_run 0) -eq 1 ] && echo $(uci_get_by_type global auto_run_time_m)' '$(uci_get_by_type global auto_run_time_h)' * * * /usr/share/jd-dailybonus/newapp.sh -w' >>$CRON_FILE
-    [ $(uci_get_by_type global auto_update 0) -eq 1 ] && echo '1 '$(uci_get_by_type global auto_update_time)' * * * /usr/share/jd-dailybonus/newapp.sh -u' >>$CRON_FILE
+    [ $(uci_get_by_type global auto_run 0) -eq 1 ] && echo $(uci_get_by_type global auto_run_time_m)' '$(uci_get_by_type global auto_run_time_h)' * * * sh /usr/share/jd-dailybonus/newapp.sh -w' >>$CRON_FILE
+    [ $(uci_get_by_type global auto_update 0) -eq 1 ] && echo '1 '$(uci_get_by_type global auto_update_time)' * * * sh /usr/share/jd-dailybonus/newapp.sh -u' >>$CRON_FILE
     crontab $CRON_FILE
     /etc/init.d/cron restart
 }

